File: ployglots/mta-ployglot-pkg-app-yaml-overwrite/packages/applier-py/src/app_yaml_from_context/default_resolver.py
"""Lazily-imported default resolver from runtime_template_resolver."""

from __future__ import annotations

from typing import Any

_ENGINE_PKG = "runtime_template_resolver"
_cached: Any = None


def default_resolver() -> Any:
    global _cached
    if _cached is not None:
        return _cached
    try:
        import importlib

        mod = importlib.import_module(_ENGINE_PKG)
    except ImportError as exc:
        raise RuntimeError(
            f"applier requires the '{_ENGINE_PKG}' package to be installed "
            f"when no resolver is injected. Install it or pass resolver= explicitly. "
            f"(cause: {exc})"
        ) from exc
    if not hasattr(mod, "create_resolver"):
        raise RuntimeError(
            f"'{_ENGINE_PKG}' did not expose create_resolver(); pass resolver= explicitly."
        )
    _cached = mod.create_resolver()
    return _cached

File: ployglots/mta-ployglot-pkg-app-yaml-overwrite/packages/applier-mjs/src/default-resolver.mjs
const ENGINE_PKG = '@ployglot/runtime-template-resolver';

let _cached;

export async function defaultResolver() {
  if (_cached) return _cached;
  let mod;
  try {
    mod = await import(ENGINE_PKG);
  } catch (err) {
    throw new Error(
      `applier requires the '${ENGINE_PKG}' package to be installed when no resolver is injected. ` +
        `Install it or pass options.resolver explicitly. (cause: ${err?.message ?? err})`,
    );
  }
  if (typeof mod.createResolver !== 'function') {
    throw new Error(
      `'${ENGINE_PKG}' did not expose createResolver(); pass options.resolver explicitly.`,
    );
  }
  _cached = mod.createResolver();
  return _cached;
}
